pkg/koordlet/statesinformer/impl/states_noderesourcetopology.go

```diff
@@ -785,18 +787,42 @@ func (s *nodeTopoInformer) calTopologyZoneList(nodeCPUInfo *metriccache.NodeCPUI
 		} else {
 			cpuQuant = resource.MustParse("0")
 		}
+
+		hugepage2Mbyte := uint64(0)
+		hugepage1Gbyte := uint64(0)
+		hugepage2MQuant := *resource.NewQuantity(0, resource.BinarySI)
+		hugepage1GQuant := *resource.NewQuantity(0, resource.BinarySI)
+
+		if isReportEnabled {
```

Comment:
How about directly check `if features.DefaultKoordletFeatureGate.Enabled(features.HugePageReport) {}`?
And perhaps we can move the hugepage calculation after L811